services/headless-lms/models/src/library/progressing.rs

```diff
@@ -48,19 +49,66 @@ pub async fn update_automatic_completion_status_and_grant_if_eligible(
             submodule_completions_required,
         )
         .await?;
+
+        if let Some(thresholds) = suspected_cheaters::get_thresholds_by_id(conn, course_instance_id)
+            .await
+            .optional()?
+        {
+            //let average_duration_seconds = course_instances::get_course_average_duration(conn, course_instance_id).await?;
+
+            check_and_insert_suspected_cheaters(
```

Comment:
Use a transaction here and use the same transaction when calling create_automatic_course_module_completion_if_eligible